tests/test_integrated_channels/test_canvas/test_client.py

```diff
@@ -86,3 +90,154 @@ def test_client_instantiation_fails_without_refresh_token(self):
             canvas_api_client = CanvasAPIClient(self.enterprise_config)
             canvas_api_client._create_session()  # pylint: disable=protected-access
         assert client_error.value.__str__() == "Failed to generate oauth access token: Refresh token required."
+
+    def test_course_delete_fails_with_empty_data(self):
+        self.transmission_with_empty_data("delete_content_metadata")
+
+    def test_course_update_fails_with_empty_data(self):
+        self.transmission_with_empty_data("update_content_metadata")
+
+    def transmission_with_empty_data(self, request_type):
```

Comment:
I get why you structured the helpers in this way, but I think it'd be better if they were all together at the top of the file. Python allows for functions to be defined in any order, but it's still a bit weird to have the helpers after the functions calling them. It's also a little weird seeing them mixed in between tests.

Reply:
fair points, I can reorder.